VehicleUI/src/main/java/com/drivequant/drivekit/vehicle/ui/vehicledetail/fragment/VehicleDetailFragment.kt

```diff
@@ -78,14 +81,41 @@ class VehicleDetailFragment : Fragment() {
     private var hasChangesToUpdate = false
 
     private var imageView: ImageView? = null
-    @DrawableRes private var vehicleDrawableId: Int? = null
 
-    private var imageUri: Uri? = null
+    private var cameraImageFilePath: String? = null
 
     private lateinit var onCameraCallback: OnCameraPictureTakenCallback
     private lateinit var menu: Menu
     private lateinit var alert: AlertDialog
 
+    private lateinit var pickMedia: ActivityResultLauncher<PickVisualMediaRequest>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        registerGalleryLauncher()
+    }
+
+    private fun registerGalleryLauncher() {
+        // Registers a photo picker activity launcher in single-select mode.
+        this.pickMedia = registerForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+            // Callback is invoked after the user selects a media item or closes the
+            // photo picker.
+            val vehicleId = vehicleId
+            if (uri != null && vehicleId != null) {
+                DriveKitVehicleUI.coroutineScope.launch {
```

Comment:
Je pense qu'il n'y a pas besoin de créer un scope et qu'on peut utiliser celui du Fragment :
```suggestion
                lifecycleScope.launch {
```
Comme ça si le Fragment disparaît, on ne fait pas de traitement inutile.

J'ai juste un doute, faut-il remplacer `if (success) {` par `if (success && isActive) {` ?

Même chose plus bas, ligne 214.

Reply:
Après échanges : 
* utilisation du `scope` du Fragment
* pas de vérification sur `isActive`